EUR-Lex.js

```diff
@@ -105,112 +115,175 @@ function doWeb(doc, url) {
+		
+		var passedBy = doc.querySelectorAll('meta[property="eli:passed_by"]');
+		var passedByArray = [];
+		for (let i = 0; i < passedBy.length; i++) {
+			passedByArray.push(passedBy[i].getAttribute("resource").split("/").pop());
+		}
+		item.legislativeBody = passedByArray.join(", ");
+		
+		item.url = attr(doc, 'meta[typeOf="eli:LegalResource"]', "about") + "/" + language.toLowerCase();
+		// eli:is_about -> eurovoc -> tags
 	}
-	
-	item.number = attr(doc, 'meta[property="eli:id_local"]', 'content');
-	
-	item.date = attr(doc, 'meta[property="eli:date_publication"]', 'content');
-	// attr(doc, 'meta[property="eli:date_document"]', 'content');
+	else if (docSector == "6") {
+		// type: case
+		// pretty hacky stuff, as there's little metadata available
+		var docCourt = docType.substr(0, 1);
+		if (docCourt == "C") {
+			item.court = languageMapping[languageUrl][1] || languageMapping["EN"][1];
+		}
+		else if (docCourt == "T") {
+			item.court = languageMapping[languageUrl][2] || languageMapping["EN"][2];
+		}
+		else if (docCourt == "F") {
+			item.court = languageMapping[languageUrl][3] || languageMapping["EN"][3];
+		}
+		item.url = url;
 
-	var passedBy = doc.querySelectorAll('meta[property="eli:passed_by"]');
-	var passedByArray = [];
-	for (let i=0; i<passedBy.length; i++) {
-		passedByArray.push(passedBy[i].getAttribute('resource').split('/').pop());
+		if (docType.substr(1) == "J") { // Judgments
+			var titleParts = attr(doc, 'meta[name="WT.z_docTitle"]', "content").replace(/\./g,"").split("#");
+			item.caseName = titleParts[1];
+			item.abstractNote = titleParts[titleParts.length - 2];
+			item.docketNumber = titleParts.pop();
+			item.dateDecided = titleParts[0].substr(titleParts[0].search(/[0-9]/g));
+		}
+		else { // Orders, summaries, etc.
+			item.caseName = attr(doc, 'meta[name="WT.z_docTitle"]', "content").replace(/#/g," ");
+			item.dateDecided = celex.substr(1, 4);
+		}
 	}
-	item.legislativeBody = passedByArray.join(', ');
-	
-	item.url = attr(doc, 'meta[typeOf="eli:LegalResource"]', 'about') + '/' + language;
-	
-	// eli:is_about -> eurovoc -> tags
+	else {
+		// type: bill
+		item.title = attr(doc, 'meta[name="WT.z_docTitle"]', "content");
+		item.date = celex.substr(1, 4);
+		item.url = url;
+		if (docType == "C") {
+			var celexCParts = celex.substr(1).split("/");
+			item.code = "OJ C";
+			item.codeVolume = celexCParts[1];
+			item.codePages = celexCParts[2];
+		}
+	}
+	// attachments
+	// type: all
+	var pdfurl = "https://eur-lex.europa.eu/legal-content/" + languageUrl + "/TXT/PDF/?uri=CELEX:" + celex;
+	var htmlurl = "https://eur-lex.europa.eu/legal-content/" + languageUrl + "/TXT/HTML/?uri=CELEX:" + celex;
+	item.attachments = [{ url: pdfurl, title: "EUR-Lex PDF (" + languageUrl + ")", mimeType: "application/pdf" }];
+	item.attachments.push({ url: htmlurl, title: "EUR-Lex HTML (" + languageUrl + ")", mimeType: "text/html" });
```

Comment:
Can this not just be a snapshot of the page the user is translating? Either way, should add `snapshot: true`.